### ISSUE TITLE
Autoconfigure IPTV Simple Client fails

### ISSUE THREAD

**Issue by bigjim80** [add-ons/service.iptv.manager#107]
I have a Nokia 8010 streaming box with Android 11. I did a fresh install with the latest Kodi.
I exactly followed the installation guide for IPTV Manager
When I try to autoconfigure IPTV Simple, I got a successful confirmation.
However, IPTV Simple is not configured at all. I get the message EPG file path is not configured.
This is the log when running the autoconfigure IPTV Simple:
[https://ibb.co/ThX4b9S](url)

Any idea?
Thanks!

**Comment by peno64** [comment]
Zie https://www.facebook.com/groups/kodivlaanderen voor een work-around

**Comment by bigjim80** [comment]
@peno64 Thanks for the update! On Android 11 there is the additional issue that you can't perform this workaround. The dialog for choosing the paths in IPTV Simple does not allow you to browse to the Kodi data folder (I guess this is due to increased security on Android 11)

However I found that IPTV Simple is using a config file named "instance-settings-1.xml" instead of "settings.xml". IPTV Manager is writing into the last one.

I hope this can help for a bugfix.

I've renamed settings.xml to instance-settings-1.xml (overwrite) and now it works!

**Comment by peno64** [comment]
@bigjim80 On kodi 19 the used file is also instance-settings-1.xml and there it works so that is why I kept this name.

Overwriting this file could work also but is not something everybody can do...

**Comment by bigjim80** [comment]
Thx for the info! I mentioned the overwrite method just as another workaround.
So the bug is that on Kodi 20 IPTV Manager adjusts settings.xml instead of instance-settings-1.xml

**Comment by peno64** [comment]
I have more the impression that both files are ok because I don't have to overwrite instance-settings-1.xml
I just have to choose it...

**Comment by cyr06** [comment]
works too thanks

**Comment by gbstan** [comment]
> @peno64 Thanks for the update! On Android 11 there is the additional issue that you can't perform this workaround. The dialog for choosing the paths in IPTV Simple does not allow you to browse to the Kodi data folder (I guess this is due to increased security on Android 11)
> 
> However I found that IPTV Simple is using a config file named "instance-settings-1.xml" instead of "settings.xml". IPTV Manager is writing into the last one.
> 
> I hope this can help for a bugfix.
> 
> I've renamed settings.xml to instance-settings-1.xml (overwrite) and now it works!

Same issue. The problem is that on embarked systems such as Google TV, you can't rename settings.xml to instance-settings-1.xml!
What would be the workaround for setting Kodi and IPTV Simple on Google TV?

**Comment by michielvg** [comment]
> Same issue. The problem is that on embarked systems such as Google TV, you can't rename settings.xml to instance-settings-1.xml! What would be the workaround for setting Kodi and IPTV Simple on Google TV?

@gbstan 

I ran into the same issue as you. 
You can access the profile data via kodi itself (settings > File manager > Profile directory > addon_data > pvr.iptvsimple) and rename the files. 
I renamed instance-settings-1.xml to instance-settings-2.xml before renaming settings.xml to instance-settings-1.xml
This gave me two 'enabled' configs for iptv ... so I removed the one were it was still looking for remote paths.

Hope this helped.

**Comment by michielvg** [comment]
By the way, my first idea was to look where the settings.xml gets created:

[Here](https://github.com/add-ons/service.iptv.manager/blob/97eb6cd894408158edd82319d07a7d187c47978b/resources/lib/modules/iptvsimple.py#L48C46-L48C46)

```python
        # Configure IPTV Simple
        output_dir = kodiutils.addon_profile()
```

If there is no add-onn object provided for the addon_profile() call it falls back to xbmcaddon.Addon();
seen that we have an addon object from earlier in the code and it is used to create the settings I wanted to try to add it to the call.

```python
        # Configure IPTV Simple
        output_dir = kodiutils.addon_profile(addon)
```

I couldn't get the package installed on my chromecast to comfirm if this resolves the issue. And then I found the file manager in Kodi for the before described work around. So I gave up on installing a custom build.

Perhaps someone else can see if I was thinking in the right direction.

WKR

**Comment by locness3** [comment]
> By the way, my first idea was to look where the settings.xml gets created:
> 
> [Here](https://github.com/add-ons/service.iptv.manager/blob/97eb6cd894408158edd82319d07a7d187c47978b/resources/lib/modules/iptvsimple.py#L48C46-L48C46)
> 
> ```python
>         # Configure IPTV Simple
>         output_dir = kodiutils.addon_profile()
> ```

I'm new to Kodi and as far as I understand, output_dir seems to be used in order to write the playlist and epg to the `service.iptv.manager` directory. Actual configuration of IPTV Simple seems to be done via `addon.setSetting()` which I assume writes to `pvr.iptvsimple/settings.xml` as a default.

**Comment by locness3** [comment]
@peno64 Thanks but please consider contributing a fix for this issue to IPTV Manager directly, if you can.

**Comment by locness3** [comment]
Nice.

**Comment by peno64** [comment]
The addon is updated but I have no idea how to push it to the standard kodi repo. @michaelarnauts can you provide me information how to do this?

**Comment by peno64** [comment]
Release v0.2.4 can be installed manually to have this fix

**Comment by stemy2** [comment]
> Release v0.2.4 can be installed manually to have this fix

No, it can't, because the dependancy xbmc.python version 2.26 can't be satisfied.

**Comment by peno64** [comment]
@stemy2 
I guess you are on kodi 19 or newer. Did you then use the zip [service.iptv.manager-0.2.4+matrix.1.zip](https://github.com/add-ons/service.iptv.manager/releases/download/v0.2.4/service.iptv.manager-0.2.4+matrix.1.zip) ?

**Comment by stemy2** [comment]
Ok, thanks.
I installed the 0.2.4 and followed the procedure again, but it didn't fixed anything, the channels list in the TV section is still empty.

**Comment by peno64** [comment]
@stemy2 
Did you open IPTV Manager such that it opens the settings, then in IPTV Simple, did you choose Configure IPTV Simple auotmtically and then in Channels did you choose Refresh channels and guide now and then restarted Kodi?

**Comment by stemy2** [comment]
I forgot to reboot, it's working now.

**Comment by emilic14** [comment]
Hello,

Just sharing my fix with the Nvidia Shield and the IPTV simple client was not working...
The issue : the local file configuration m3u + epg was not configured and bigger issue, I could not explore the folders to point to the local folder through the kodi interface... 

Solution : modify the instance-settings-1.xml file by hand using adb in order to add the m3u8 and epg file location.
the solution is partially shared [here](https://github.com/Catch-up-TV-and-More/plugin.video.catchuptvandmore/issues/1173)
for nvidia shield you just have to do it by hand...
I recommend changing the settings first in the interface to point to a random file (so that you know then which lines to modify), Quit Kodi, recover your file through adb.
modify it with your pc : 
special://userdata/addon_data/service.iptv.manager/playlist.m3u8 
special://userdata/addon_data/service.iptv.manager/epg.xml
